### PR TITLE
Refactors transit tubes

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Refactors transit tubes to a pretty significant degree. Player-wise, they may be a tad slower, but they're gonna hopefully be a lot easier to work with map-wise and hopefully code-wise.

On the mapping side, mapping in transit tubes becomes significantly easier. I don't even want to think about how much fun it would have been in the past to map a new set of transit tubes, since you had to do all kinds of weird icon state stuff. Every directional icon was baked into an icon file, and you even had to add corners yourselves on diagonals.
Now, you just set the right dir for tubes, of which there are now plenty of helper subtypes for curves, diagonals, reversed pipes, etc. Code will automatically add the junction bits/decorations for corners and joints on init, even though they may not be immediately visible in a map editor.

On the coding side, transit tubes now no longer use sleeps or timers during their main movement loop, and instead use a small subsystem that fires once per tick. I've tried to be clever with usage so things are only added to the processing queue if they're actively moving, which should save a fair amount of processing time since tubes spend a lot of their time at stations. 

A significant amount of the heavy structural improvements to icons and code came from /tg/'s implementation.

## Why It's Good For The Game
Transit tube code is pretty crusty and old (most of it hasn't been touched for the past 9 years or so). This should hopefully bring it up to slightly more modern standards, and hopefully make things more performant going forward.

This should also hopefully make transit tubes easier to interact with on the mapping side, and possibly make them justifiable for use on the coding side (such as possibly adding them to the RPD or something).

## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/89928798/f1d4b6e0-58a5-4093-aeca-4c8088ccc808

## Testing
Loaded in all maps, and verified that the transit tubes (as configured) worked from roundstart. Also built a few transit mechanisms from scratch and verified they worked as intended.

## Changelog
:cl:
tweak: Transit tubes are now slightly slower
tweak: You can now empty out a transit pod with a crowbar
/:cl:

### CODE REVIEW

File: code/game/objects/structures/transit_tubes/station.dm [hunk 1]
```diff
@@ -27,10 +30,38 @@
...
+/obj/structure/transit_tube/station/reverse/init_tube_dirs()
+	switch(dir)
+		if(NORTH)
+			tube_dirs = list(EAST)
+		if(SOUTH)
+			tube_dirs = list(WEST)
+		if(EAST)
+			tube_dirs = list(SOUTH)
+		if(WEST)
+			tube_dirs = list(NORTH)
```
Comment (Sirryan2002) [inline code/game/objects/structures/transit_tubes/station.dm:61]:
```suggestion
	tube_dirs = list(turn(dir, -90))
```
I think this does the same exact thing

File: code/game/objects/structures/transit_tubes/transit_tube.dm [hunk 2]
```diff
@@ -122,191 +126,153 @@
...
+/obj/structure/transit_tube/update_overlays()
+	. = ..()
+	for(var/direction in directions())
+		if(!IS_DIR_DIAGONAL(direction))
+			. += create_tube_overlay(direction)
+			continue
+		if(!(direction & NORTH))
+			continue
+
+		. += create_tube_overlay(direction ^ 3, NORTH)
```
Comment (Sirryan2002) [inline code/game/objects/structures/transit_tubes/transit_tube.dm:145]:
```suggestion
		. += create_tube_overlay(direction ^ (NORTH|SOUTH), NORTH)
```

File: code/game/objects/structures/transit_tubes/transit_tube.dm [hunk 3]
```diff
@@ -122,191 +126,153 @@
...
+		if(direction & EAST)
+			. += create_tube_overlay(direction ^ 12, EAST)
```
Comment (Sirryan2002) [inline code/game/objects/structures/transit_tubes/transit_tube.dm:147]:
```suggestion
			. += create_tube_overlay(direction ^ (EAST|WEST), EAST)
```

File: code/game/objects/structures/transit_tubes/transit_tube.dm [hunk 4]
```diff
@@ -122,191 +126,153 @@
...
+		else
+			. += create_tube_overlay(direction ^ 12, WEST)
```
Comment (Sirryan2002) [inline code/game/objects/structures/transit_tubes/transit_tube.dm:149]:
```suggestion
			. += create_tube_overlay(direction ^ (EAST|WEST), WEST)
```

File: code/game/objects/structures/transit_tubes/station.dm [hunk 5]
```diff
@@ -27,10 +30,30 @@
...
+	switch(dir)
+		if(NORTH)
+			tube_dirs = list(EAST, WEST)
+		if(SOUTH)
+			tube_dirs = list(EAST, WEST)
+		if(EAST)
+			tube_dirs = list(NORTH, SOUTH)
+		if(WEST)
+			tube_dirs = list(NORTH, SOUTH)
+	boarding_dir = reverse_direction(dir)
```
[the remark this answers is not in the document]
Comment (lewcc) [reply code/game/objects/structures/transit_tubes/station.dm:46]: it does now

File: code/game/objects/structures/transit_tubes/station.dm [hunk 6]
```diff
@@ -152,10 +170,105 @@
...
+	name = "station tube pod dispenser"
+	icon_state = "open_dispenser0"
+	desc = "The lynchpin of a GOOD transit system."
+	enter_delay = 1
+	base_icon_state = "dispenser0"
+	hatch_state = TRANSIT_TUBE_OPEN
+
+
+
+/obj/structure/transit_tube/station/dispenser/examine(mob/user)
```
Comment (Henri215) [inline code/game/objects/structures/transit_tubes/station.dm:211]:
```suggestion
// Special dispenser station, it creates a pod for you to enter when you bump into it.
/obj/structure/transit_tube/station/dispenser
	name = "station tube pod dispenser"
	desc = "The lynchpin of a GOOD transit system."
	icon_state = "open_dispenser0"
	enter_delay = 1
	base_icon_state = "dispenser0"
	hatch_state = TRANSIT_TUBE_OPEN

/obj/structure/transit_tube/station/dispenser/examine(mob/user)
```

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm [hunk 7]
```diff
@@ -18,76 +37,129 @@
...
+
+/obj/structure/transit_tube_pod/proc/outside_tube()
+	var/list/savedcontents = contents.Copy()
+	var/saveddir = dir
+	var/turf/destination = get_edge_target_turf(src, saveddir)
+	visible_message("<span class='warning'>[src] ejects its insides out!</span>")
+	deconstruct(FALSE)//we automatically deconstruct the pod
+	for(var/i in savedcontents)
+		var/atom/movable/AM = i
+		AM.throw_at(destination,rand(1,3),5)
```
Comment (Henri215) [inline code/game/objects/structures/transit_tubes/transit_tube_pod.dm:96]:
```suggestion
		AM.throw_at(destination, rand(1, 3), 5)
```

File: code/game/objects/structures/transit_tubes/transit_tube_pod.dm [hunk 8]
```diff
@@ -1,3 +1,6 @@
+#define MOVE_ANIMATION_STAGE_ONE 1
+#define MOVE_ANIMATION_STAGE_TWO 2
```
Comment (Henri215) [inline code/game/objects/structures/transit_tubes/transit_tube_pod.dm:2]: These are not being undefined.

**Review by S34NW — APPROVED** [review]

**Pull request merged**